gtsfm/pose_slam/pose_slam.py

```diff
@@ -0,0 +1,85 @@
+"""Pose SLAM initialization module from relative pose priors.
+
+Authors: Akshay Krishnan, Frank Dellaert
+"""
+from typing import Dict, List, Optional, Tuple
+
+import dask
+import gtsam
+from dask.delayed import Delayed
+from gtsam import Pose3
+
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.pose_prior import PosePrior
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+
+PRIOR_NOISE_SIGMAS = [0.001, 0.001, 0.001, 0.1, 0.1, 0.1]
+POSE_PRIOR_NOISE = gtsam.noiseModel.Diagonal.Sigmas(PRIOR_NOISE_SIGMAS)
+
+logger = logger_utils.get_logger()
+
+
+class PoseSlam:
+    def run_pose_slam(
+        self,
+        num_images: int,
+        relative_pose_priors: Dict[Tuple[int, int], PosePrior] = {},
```

Comment:
Optional arguments are dangerous. Also, we would have empty list if loader doesnt have priors.